### PR TITLE
Metadata: Update values to strings

### LINKED ISSUE #771: Metdata: Inconsistent handling of values
## Summary
Metadata values are handled differently depending on how the metadata is sent to the system.

1. Pipeline Metadata and Metadata imported through the GraphQL API
   * Types in metadata are preserved
      * numbers not quoted are saved as numbers
      * true values are saved as booleans
      * false values are not saved instead that field if present is removed
      * nil values are not saved instead that field if present is removed
2. Metadata addition or updates through the UI, or Metadata imported from a file
   * Values are always saved as strings
   * Empty strings, nil values are not saved instead that field if present is removed

## Potential Solutions
1. Ignore type and transform values into strings before storing
~~2. Auto-detect type from metadata values when manually adding/updating or importing through the UI~~
~~3. Allow the user to specify the type for each file being added/updated or imported through the UI~~

### PR DESCRIPTION
**Pull request by ksierks**
## What does this PR do and why?
Converting metadata values to strings, so we can search all metadata values. (See #807.)
Fixes #771.

## Screenshots or screen recordings
image

## How to set up and validate locally
**Add metadata through the UI:**
1. Navigate to a samples page that the logged in user has access to update.
5. Hit the `Import Metadata` button.
6. Select a file. (See /irida-next-core/test/fixtures/files/metadata/valid.xlsx)
7. Select the `Sample ID` Column.
8. Hit the `Import Metadata` button.
9. Verify the sample metadata was added as strings.
10. Follow the `Test searching metadata` instructions below.

**Add metadata through GraphQL:**
1. Navigate to `http://localhost:3000` and login as admin.
2. Navigate to `http://localhost:3000/graphiql`.
3. Run the following to retrieve a project:
```
query getProject{
  project(fullPath: "bacillus/bacillus-anthracis/outbreak-2022") {
    id
    name
    path
  }
}
```
4. Run the following to create a fresh sample:
```
mutation createSample{
  createSample(input: { 
    projectId: "<insert here>", 
    name: "Sample Name", 
    description: "Sample Description" 
  }) {
    sample {
      id,
      puid,
      name,
      description
    }
    errors {
      path,
      message
    }
  }
}
```
5. Run the following to add metadata to the sample:
```
mutation addMetadataToSample{
  updateSampleMetadata(input: {
    sampleId: "<insert here>", 
    metadata: { true_boolean: true, false_boolean: false, nil: null, empty: "", number: 123, date: "2024-03-11" }
  }) {
    sample{
      id,
      name,
      description,
      metadata
    },
    status,
    errors {
      path
      message
    }
  }
}
```
6. Verify the date, number, true_boolean, and false_boolean were added successfully as strings.
7. Follow the `Test searching metadata` instructions below.

**Test searching metadata**
1. Navigate to a samples page.
2. Toggle metadata.
3. Search using the format FIELD_NAME:FIELD_VALUE.
4. Verify the results when searching for dates, numbers, booleans, etc. 
 
## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/services/samples/metadata/update_service.rb [hunk 1]
```diff
@@ -69,14 +67,10 @@ def validate_metadata_param
...
-      def transform_metadata_keys
-        # Without transforming and downcasing keys,
-        # issues with overwritting can occur and multiples of the same key can appear
-        @metadata = @metadata.transform_keys { |key| key.to_s.downcase }
-      end
-
       def perform_metadata_update
         @metadata.each do |key, value|
+          key = key.to_s.downcase
+          value = value.to_s # remove data types
```
Comment (JeffreyThiessen) [inline app/services/samples/metadata/update_service.rb:73]: Can you strip white space from the keys and values? I noticed that white space is also getting included when trying to look up the sample id from an xlsx file.

**Review by JeffreyThiessen — CHANGES_REQUESTED** [review]
Overall it looks good. Just one change, and add a test to verify it as well.

**Review by JeffreyThiessen — APPROVED** [review]
Looks great!

**Review by ChrisHuynh333 — APPROVED** [review]
Looks good!

**Review by deepsidhu85 — APPROVED** [review]
Looks good to me! Thanks for fixing up the migration

**Pull request merged**